src/ActionPreviewHeader.jsx

```diff
@@ -1,44 +1,55 @@
-import React from 'react';
+import React, { Component } from 'react';
 
-const ActionPreviewHeader = ({
-  styling, inspectedPath, onInspectPath, tab, onSelectTab
-}) => {
-  return (
-    <div key='previewHeader' {...styling('previewHeader')}>
-      <div {...styling('tabSelector')}>
-        {['Action', 'Diff', 'State'].map(t =>
-          <div onClick={() => onSelectTab(t)}
-               key={t}
-               {...styling([
-                 'selectorButton',
-                 t === tab && 'selectorButtonSelected'
-               ], t === tab)}>
-            {t}
-          </div>
-        )}
-      </div>
-      <div {...styling('inspectedPath')}>
-        {inspectedPath.length ?
-          <span {...styling('inspectedPathKey')}>
-            <a onClick={() => onInspectPath([])}
-               {...styling('inspectedPathKeyLink')}>
-              {tab}
-            </a>
-          </span> : tab
-        }
-        {inspectedPath.map((key, idx) =>
-          idx === inspectedPath.length - 1 ? key :
-          <span key={key}
-             {...styling('inspectedPathKey')}>
-            <a onClick={() => onInspectPath(inspectedPath.slice(0, idx + 1))}
-               {...styling('inspectedPathKeyLink')}>
-              {key}
-            </a>
-          </span>
-        )}
+class ActionPreviewHeader extends Component {
+  constructor(props) {
+    super(props);
+    this.tabs = ['Action', 'Diff', 'State'];
```

Comment:
I would doubt it. Even when an action is selected this component is rerendered 4 times. So, every time it will do `props.customTabs.forEach`, which also, as the result, will create a new function inside rendering 4 times, and also will invoke garbage collector every time. 

Also the same goes for arrow functions added in `onClick` bellow. Having functions inside `render` affects the performance a lot, [more details](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md).

It would be great if someone will have time to investigate and improve the performance of the components.

OK, will add it as well in `componentWillReceiveProps` if you think we'll need to change `props.customTabs` dynamically.